### DISCUSSION TITLE
Use vector drawables as markers

### DISCUSSION THREAD

**Discussion by Psijic** [mapsforge/vtm#1068] [Q&A] [answered]
I see there is `AndroidSvgBitmap` need to be set for the default marker image. But I didn't find how to use vector .xml drawables.
Suppose, using `androidx.compose.ui.graphics.vector.ImageVector` (`Icons.Default.Marker`) is something above the level here.

**Comment by devemux86** [comment]
Could convert any third-party image like the xml drawable to the supported format.

**Reply by Psijic** [reply]
Is there a sample?

**Reply by devemux86** [reply]
Can search online how to convert xml drawables to Android bitmaps.

**Reply by demiantres** [reply]
https://vd.floo.app/

**Reply by Psijic** [reply]
I need to use Android xml drawables as vector marker images, no need to convert.
It's actually the opposite: to add an xml drawable, you need to convert it FROM .svg file and then use in Android.
Samples: `Icon(Icons.Filled.Close, null)` or `Image(painterResource(R.drawable.logo))`
from `package androidx.compose.material3`

**Reply by devemux86** [reply]
VTM uses bitmaps as final format in markers on each platform (check the API).
How these bitmaps can be created is app's responsibility, from drawables, svg, etc.

If SVG are available, can use them directly with AndroidSVG library.
Same process has been done with map theme SVG images all this time, before xml drawables ever existed.

**Reply by Psijic** [reply]
So, `R.drawable.marker` as an xml vector image is not supported out of the box?

**Reply by demiantres** [reply, marked as answer]
You can google for "android xml to bitmap" which yields some examples how to convert the xml to a bitmap which you can then use in VTM. For example: https://stackoverflow.com/questions/33696488/getting-bitmap-from-vector-drawable

**Reply by Psijic** [reply]
The question is in autoscale with all resolutions support without artifacts like proper vector does. If it exists - that's fine.
